samples/crypto/test_sample/README.rst

```diff
@@ -0,0 +1,258 @@
++--------------------+-------------+----------------------------------------------------------------------------+----------------------------------------------------------------------------+
+|                    | HMAC SHA512 |                                                                            | `NIST - HMAC`_                                                             |
++--------------------+-------------+----------------------------------------------------------------------------+----------------------------------------------------------------------------+
+| HKDF               | HKDF SHA256 | `RFC 5869 - HMAC-based Extract-and-Expand Key Derivation Function (HKDF)`_ | `RFC 5869 - HMAC-based Extract-and-Expand Key Derivation Function (HKDF)`_ |
++--------------------+-------------+----------------------------------------------------------------------------+----------------------------------------------------------------------------+
+|                    | HKDF SHA512 |                                                                            | `RFC 5869 - HMAC-based Extract-and-Expand Key Derivation Function (HKDF)`_ |
++--------------------+-------------+----------------------------------------------------------------------------+----------------------------------------------------------------------------+
+
+
+Requirements
+************
+
+* One of the following development boards:
+
+  * |nRF9160DK|
+  * |nRF5340DK|
+  * |nRF52840DK|
+  * |nRF52DK|
+  * |nRF51DK|
+
+* something else??
+
+
+Building and running
+********************
+
+.. |sample path| replace:: :file:`samples/bluetooth/central_bas`
+
+.. include:: /includes/build_and_run.txt
+
+.. _crypto_test_testing:
+
+Testing
+=======
+
+1. Compile and program the application.
+#. Observe the result of the different test vectors in the log using an RTT viewer or a terminal emulator.
+   Make sure the test vector verdict is "passed"::
+
+      ***************************************
+      All Tests Done
+      X test vectors passed
+      0 test vectors failed
+      ***************************************
+#.
+
+Additional test cases and test vectors
+======================================
+
+Test cases and test vectors can be added to the test suite either by including additional source files or by extending the existing files.
+
+
+Test case
+---------
+
+A test case is a function designed to verify parts of the functionality of an nrf_crypto operation.
+Most nrf_crypto operations, like hash calculations and ECDH, have multiple test cases to be able to cover all features.
+A typical test case loops over all related test vectors, and logs the verdict of each.
+
+An example of a hash test case in a simplified form is shown below::
+
+   NRF_SECTION_DEF(test_vector_hash_data, test_vector_hash_t);
```

Comment:
/**@brief Function for the test execution.
 */
void exec_test_case_hmac(void)
{
	int err_code = -1;

	/* Initialize the HMAC module. */
	mbedtls_md_init(&md_context);

	const mbedtls_md_info_t *p_md_info =
		mbedtls_md_info_from_type(p_test_vector->digest_type);
	err_code = mbedtls_md_setup(&md_context, p_md_info, 1);
	if (err_code != 0) {
		LOG_WRN("mb setup ec: -0x%02X", -err_code);
	}
	TEST_VECTOR_ASSERT_EQUAL(err_code, 0);

	start_time_measurement();
	err_code = mbedtls_md_hmac_starts(&md_context, m_hmac_key_buf, key_len);
	TEST_VECTOR_ASSERT_EQUAL(err_code, 0);

	err_code =
		mbedtls_md_hmac_update(&md_context, m_hmac_input_buf, in_len);
	TEST_VECTOR_ASSERT_EQUAL(err_code, 0);

	/* Finalize the HMAC computation. */
	err_code = mbedtls_md_hmac_finish(&md_context, m_hmac_output_buf);
	stop_time_measurement();

	TEST_VECTOR_ASSERT_EQUAL(p_test_vector->expected_err_code, err_code);

	/* Verify the generated HMAC. */
	TEST_VECTOR_ASSERT_EQUAL(expected_hmac_len, hmac_len);
	TEST_VECTOR_MEMCMP_ASSERT(m_hmac_output_buf, m_hmac_expected_output_buf,
				  expected_hmac_len,
				  p_test_vector->expected_result,
				  "Incorrect hmac");

	mbedtls_md_free(&md_context);
}

ITEM_REGISTER(test_case_hmac_data, test_case_t test_hmac) = {
	.p_test_case_name = "HMAC",
	.setup = hmac_setup,
	.exec = exec_test_case_hmac,
	.teardown = hmac_teardown,
	.vector_type = TV_HMAC,
	.vectors_start = __start_test_vector_hmac_data,
	.vectors_stop = __stop_test_vector_hmac_data,
};

Reply:
done

Reply:
Closing this as there is different comment on the same content referenced here: https://github.com/nrfconnect/sdk-nrf/pull/1766#discussion_r432421526